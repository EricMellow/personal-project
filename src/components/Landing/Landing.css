.landing {
  background-image: url(../../assets/splitBackground.jpg);
  height: 85vh;
  background-size: cover;
  display: flex;
  background-repeat: no-repeat;
  justify-content: space-around;
  align-items: center;
}

.zip {
  width: 80%;
  height: 4rem;
  background: none;
  color: rgba(28, 141, 196);
  font-size: 2rem;
  padding-left: 10px;
}

.landing-form {
  width: 40vw;
  display: flex;
  height: 50%;
  background-color: rgba(255, 255, 255, 0.9);
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.landing-signup {
  width: 40vw;
  display: flex;
  height: 50%;
  background-color: rgba(28, 141, 196, .9);
  justify-content: center;
  align-items: center;
  flex-direction: column;
}

.create-text {
  color: white;
  width: 80%;
}

.enter-text {
  color: rgba(28, 141, 196, .9);
  width: 80%
}

.nav-signup {
  color: white;
  text-decoration: none;
  border: 3px solid white;
  border-radius: 5px;
  padding: 12px;
  margin: 0;
  font-size: 2rem
}