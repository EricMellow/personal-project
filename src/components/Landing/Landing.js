import React, { Component } from "react";
import './Landing.css';
import { connect } from "react-redux";
import { addZipcode } from "../../actions/zipcodeActions";
import { getLocation } from "../../apiCalls/apiCalls";
import { NavLink } from "react-router-dom";

export class Landing extends Component {
  constructor(props) {
    super(props);
    this.state = {
      zip: undefined
    };
  }

  handleInputChange = (event) => {
    event.preventDefault();
    this.setState({
      [event.target.name]: event.target.value
    });
  }

  handleSubmit = (event) => {
    event.preventDefault();
    this.props.storeZipcode(this.state.zip);
    this.resetState();
    this.props.history.push('/distance');
  }

  resetState = () => {
    this.setState({
      zip: undefined
    });
  }

  render() {
    return (
      <div className="landing">
        <form className="landing-form" onSubmit={event => this.handleSubmit(event)}>
          <input
            name="zip"
            placeholder="Zip Code"
            className="zip"
            type="number"
            value={this.state.zip}
            onChange={event => this.handleInputChange(event)}
          >
          </input>
          <h3 className="enter-text">Enter your zip code to see activities near you</h3>
        </form>
        <div className="landing-signup">
          <NavLink to="/signup" className="nav-signup">Sign Up</NavLink>
          <h3 className="create-text">To create an activity for people to join</h3>
        </div>
      </div>
    );
  }
}

export const mapDispatchToProps = (dispatch) => ({
  storeZipcode: (zipcode) => dispatch(addZipcode(zipcode))
})

export default connect(null, mapDispatchToProps)(Landing);