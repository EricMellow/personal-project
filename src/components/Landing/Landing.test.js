import { Landing } from "./Landing";
import { shallow } from "enzyme";
import React from "react";

describe('Landing', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<Landing />);
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });

  describe('handleInputChange', () => {
    
    it('should set state with the event info', () => {
      const wrapper = shallow(<Landing />);
      const mockEvent = {
        target: {
          name: 'zip',
          value: 80401,
        },
        preventDefault: jest.fn()
      };

      expect(wrapper.state('zip')).toEqual(undefined)
      wrapper.instance().handleInputChange(mockEvent)
      expect(wrapper.state('zip')).toEqual(80401)
    });
  });

  describe('render', () => {
    
    it('should call handleInputChange onChange of the input', () => {
      const wrapper = shallow(<Landing />)
      wrapper.instance().handleInputChange = jest.fn()

      wrapper.find('.zip').simulate('change')
      expect(wrapper.instance().handleInputChange).toHaveBeenCalled()
    });
  });
});