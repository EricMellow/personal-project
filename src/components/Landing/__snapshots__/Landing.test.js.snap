// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Landing should match the snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Landing />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <form
          className="landing-form"
          onSubmit={[Function]}
>
          <input
                    className="zip"
                    name="zip"
                    onChange={[Function]}
                    placeholder="Zip Code"
                    type="number"
                    value={undefined}
          />
          <h3
                    className="enter-text"
          >
                    Enter your zip code to see activities near you
          </h3>
</form>,
        <div
          className="landing-signup"
>
          <NavLink
                    activeClassName="active"
                    ariaCurrent="true"
                    className="nav-signup"
                    to="/signup"
          >
                    Sign Up
          </NavLink>
          <h3
                    className="create-text"
          >
                    To create an activity for people to join
          </h3>
</div>,
      ],
      "className": "landing",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <input
              className="zip"
              name="zip"
              onChange={[Function]}
              placeholder="Zip Code"
              type="number"
              value={undefined}
/>,
            <h3
              className="enter-text"
>
              Enter your zip code to see activities near you
</h3>,
          ],
          "className": "landing-form",
          "onSubmit": [Function],
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "zip",
              "name": "zip",
              "onChange": [Function],
              "placeholder": "Zip Code",
              "type": "number",
              "value": undefined,
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Enter your zip code to see activities near you",
              "className": "enter-text",
            },
            "ref": null,
            "rendered": "Enter your zip code to see activities near you",
            "type": "h3",
          },
        ],
        "type": "form",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <NavLink
              activeClassName="active"
              ariaCurrent="true"
              className="nav-signup"
              to="/signup"
>
              Sign Up
</NavLink>,
            <h3
              className="create-text"
>
              To create an activity for people to join
</h3>,
          ],
          "className": "landing-signup",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "activeClassName": "active",
              "ariaCurrent": "true",
              "children": "Sign Up",
              "className": "nav-signup",
              "to": "/signup",
            },
            "ref": null,
            "rendered": "Sign Up",
            "type": [Function],
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "To create an activity for people to join",
              "className": "create-text",
            },
            "ref": null,
            "rendered": "To create an activity for people to join",
            "type": "h3",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <form
            className="landing-form"
            onSubmit={[Function]}
>
            <input
                        className="zip"
                        name="zip"
                        onChange={[Function]}
                        placeholder="Zip Code"
                        type="number"
                        value={undefined}
            />
            <h3
                        className="enter-text"
            >
                        Enter your zip code to see activities near you
            </h3>
</form>,
          <div
            className="landing-signup"
>
            <NavLink
                        activeClassName="active"
                        ariaCurrent="true"
                        className="nav-signup"
                        to="/signup"
            >
                        Sign Up
            </NavLink>
            <h3
                        className="create-text"
            >
                        To create an activity for people to join
            </h3>
</div>,
        ],
        "className": "landing",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <input
                className="zip"
                name="zip"
                onChange={[Function]}
                placeholder="Zip Code"
                type="number"
                value={undefined}
/>,
              <h3
                className="enter-text"
>
                Enter your zip code to see activities near you
</h3>,
            ],
            "className": "landing-form",
            "onSubmit": [Function],
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "zip",
                "name": "zip",
                "onChange": [Function],
                "placeholder": "Zip Code",
                "type": "number",
                "value": undefined,
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Enter your zip code to see activities near you",
                "className": "enter-text",
              },
              "ref": null,
              "rendered": "Enter your zip code to see activities near you",
              "type": "h3",
            },
          ],
          "type": "form",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <NavLink
                activeClassName="active"
                ariaCurrent="true"
                className="nav-signup"
                to="/signup"
>
                Sign Up
</NavLink>,
              <h3
                className="create-text"
>
                To create an activity for people to join
</h3>,
            ],
            "className": "landing-signup",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "activeClassName": "active",
                "ariaCurrent": "true",
                "children": "Sign Up",
                "className": "nav-signup",
                "to": "/signup",
              },
              "ref": null,
              "rendered": "Sign Up",
              "type": [Function],
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "To create an activity for people to join",
                "className": "create-text",
              },
              "ref": null,
              "rendered": "To create an activity for people to join",
              "type": "h3",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
